llarp/nodedb.hpp

```diff
@@ -96,8 +96,7 @@ namespace llarp
       for (const auto& entry : m_Entries)
         keys.emplace_back(entry.first);
 
-      std::shuffle(
-          keys.begin(), keys.end(), std::mt19937{static_cast<unsigned int>(llarp::randint())});
+      std::shuffle(keys.begin(), keys.end(), std::mt19937_64{llarp::randint()});
```

Comment:
`std::mt19937_64` construction is quite heavy (but drawing from it is very fast).  A better approach here would be to have a global, thread_local mt19937_64 around in some header:

```c++
inline thread_local std::mt19937_64 rng{std::random_device{}()};
```
or header/cpp pair:
```c++
extern thread_local std::mt19937_64 rng;
```
```c++
thread_local std::mt19937_64 rng{std::random_device{}()};
```
and then just use this directly wherever non-crypto psuedo-rng is needed (like here).  Most compilers will defer the construction until first use of the variable within a thread, *not* on every thread construction.

Reply:
`llarp::randint()` might also work fine here, but I'm not sure how soon that is available.

Reply:
Actually, perhaps a better approach is just to implement a [URNG-implementing class](https://en.cppreference.com/w/cpp/named_req/UniformRandomBitGenerator) around llarp::randint().  I think this would do it:

```C++
struct RNG {
  using result_type = uint64_t;
  static constexpr uint64_t min() { return std::numeric_limits<uint64_t>::min(); }
  static constexpr uint64_t max() { return std::numeric_limits<uint64_t>::max(); }
  uint64_t operator() { return llarp::randint(); }
};
```